src/metabase/query_processor/pivot.clj

```diff
@@ -0,0 +1,20 @@
+(ns metabase.query-processor.pivot
+  "Pivot table actions for the query processor"
+  (:require [metabase.util.i18n :refer [trs]]))
+
+(defn- generate-breakouts
+  "Generate the combinatorial breakouts for a given query pivot table query"
+  [breakouts]
+  (when (not= 3 (count breakouts))
```

Comment:
I don't see why we have to enforce 3 breakouts. Having a total seems useful for 1-oo breakouts